### PR TITLE
doc: Array of Structure Query Code sample

### PR DESCRIPTION
**Pull request by farhan0102**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          QueryJobConfiguration.newBuilder(query)
+              .setUseLegacySql(false)
+              .addNamedParameter(
+                  "arrayOfStructField",
+                  QueryParameterValue.array(arrayOfStructs.toArray(), StandardSQLTypeName.STRING))
+              .build();
+      TableResult results = bigquery.query(queryConfig);
+      results
+          .iterateAll()
+          .forEach(row -> row.forEach(val -> System.out.printf("%s", val.toString())));
```
Comment (Neenu1995) [inline samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java:61]: Check if this is printing the expected value.

File: samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (Neenu1995) [inline samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java:2]: Update License Year

File: samples/snippets/src/test/java/com/example/bigquery/QueryWithArrayOfStructsNamedParametersIT.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (Neenu1995) [inline samples/snippets/src/test/java/com/example/bigquery/QueryWithArrayOfStructsNamedParametersIT.java:2]: Update license year

Comment (farhan0102) [reply samples/snippets/src/test/java/com/example/bigquery/QueryWithArrayOfStructsNamedParametersIT.java:2]: Done!

File: samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java (hunk 2 above)
Comment (farhan0102) [reply samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java:2]: Done

File: samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java (hunk 1 above)
Comment (farhan0102) [reply samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java:61]: did you want me to add it as a part of the unit test?

currently it prints:
```
[INFO] Running com.example.bigquery.QueryWithArrayOfStructsNamedParametersIT
Mar 01, 2023 11:45:03 PM com.example.bigquery.QueryWithArrayOfStructsNamedParametersIT tearDown
INFO: 
FieldValue{attribute=REPEATED, value=[FieldValue{attribute=PRIMITIVE, value=QueryParameterValue{value=null, arrayValuesInner=null, structValuesInner={stringField=QueryParameterValue{value=test-stringField, arrayValuesInner=null, structValuesInner=null, type=STRING, arrayType=null, structTypesInner=null}}, type=STRUCT, arrayType=null, structTypesInner={stringField=QueryParameterValue{value=test-stringField, arrayValuesInner=null, structValuesInner=null, type=STRING, arrayType=null, structTypesInner=null}}}}]}Query with Array of struct parameters performed successfully.

[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.319 s - in com.example.bigquery.QueryWithArrayOfStructsNamedParametersIT
```

Comment (Neenu1995) [reply samples/snippets/src/main/java/com/example/bigquery/QueryWithArrayOfStructsNamedParameters.java:61]: This is good. LGTM

**Review by Neenu1995 — APPROVED** [review]

**Comment by Neenu1995** [comment]
nit: The conventional commit for samples is doc. `feat` will add it as a feature to the release notes.

**Pull request merged**